include/circt/Dialect/FIRRTL/FIRRTLExpressions.td

```diff
@@ -15,7 +15,7 @@ def APSIntAttr : Attr<CPred<"$_self.isa<::mlir::IntegerAttr>()">,
   let storageType = [{ ::mlir::IntegerAttr }];
   let returnType = [{ ::llvm::APSInt }];
   let constBuilderCall = "IntegerAttr::get($_builder.getContext(), $0)";
-  let convertFromStorage = "APSInt($_self.getValue(), !getType().isSigned())";
+  let convertFromStorage = "APSInt($_self.getValue(), !isSigned())";
```

Comment:
The `APSIntAttr` is a little weird.  It doesn't store the sign bit, instead it gets it from the return type of the ConstantOp it is embedded in.  Should this actually be getting the sign from the type of the `IntegerAttr`?

Reply:
Looks like the IntegerAttr has a function we can use: `APSInt IntegerAttr::getAPSInt()`